### PR TITLE
feat: Author link press on article byline

### PR DESCRIPTION
**Pull request by tuncaulubilge**
Added onAuthorPress callback to article byline links for author names. This callback will be implemented in native to navigate to author profiles.

### CODE REVIEW

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/46abd02e3d9f1bdd4e78ca56487241c692b7d501/index.html

File: packages/article-byline/article-byline.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ const ArticleByline = ({ ast, style }) =>
           style={[linkStyles.link, style.link]}
           key={key}
           url={url}
-          onPress={() => {}}
+          onPress={e => onAuthorPress(e, { slug: attributes.slug, url })}
```
Comment (fourlastor) [inline packages/article-byline/article-byline.js]: do we need `e` passed to `onAuthorPress`?

Comment (tuncaulubilge) [reply packages/article-byline/article-byline.js]: Removed it.
Also removed from onRelatedArticlePress for consistency

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/537c58a6aa37f51fd9a4a6987a2e8ac201b8b135/index.html

**Pull request merged**